src/routes/webhook/repositories/webhook.entity.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  Entity,
+  Column,
+  PrimaryGeneratedColumn,
+  BaseEntity,
+  Generated,
+} from 'typeorm';
+import { TxServiceEvent } from '../../events/event.dto';
+import { SendEventTypes, WebhookPublicDto } from '../dtos/webhook.dto';
+
+@Entity()
+export class Webhook extends BaseEntity {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column({ unique: true })
+  @Generated('uuid')
+  public_id: string;
+
+  @Column()
+  url: string;
+
+  @Column()
+  description: string;
+
+  @Column({ default: true })
+  isActive: boolean;
+
+  @Column()
+  authorization: string;
+
+  @Column('bigint', { array: true, default: [] })
+  chains: string[];
+
+  @Column({ default: true })
+  sendConfirmations: boolean;
+
+  @Column({ default: true })
+  sendMultisigTxs: boolean;
+
+  @Column({ default: true })
+  sendEtherTransfers: boolean;
+
+  @Column({ default: true })
+  sendTokenTransfers: boolean;
+
+  @Column({ default: true })
+  sendModuleTransactions: boolean;
+
+  @Column({ default: true })
+  sendSafeCreations: boolean;
+
+  @Column({ default: true })
+  sendMessages: boolean;
+
+  @Column({ default: true })
+  sendReorgs: boolean;
+
+  @Column({ default: true })
+  sendDelegates: boolean;
+
+  /**
+   * Check if event chainId matches the one of the webhook (everything will match if webhook chains are empty). Check if event
+   * type matches the flags enabled for the webhook
+   * @param message
+   * @returns true if event is relevant.
+   */
+  isEventRelevant(message: TxServiceEvent): boolean {
+    const chainMatches: boolean =
+      this.chains.length === 0 || this.chains.includes(message.chainId);
+    return (
+      chainMatches &&
+      ((this.sendConfirmations &&
+        (message.type === 'NEW_CONFIRMATION' ||
```

Comment:
As you defined a enum for the DTO, you can use it here instead of the `str` comparison I was doing.

So the enum can be defined for the entity and imported for the DTO

Reply:
But the types are different SendEventTypes are the group types, for example SEND_TOKEN_TRANSFER include incoming and outgoing transfers.